### PR TITLE
Fix CMake duplication, update Readme

### PR DESCRIPTION
**Pull request by kstribrnAmzn**
### Description of changes
Remove duplicate and unnecessary CMake line.
Update README to add notice that default Root
CA path is relative to binary execution directory.

### Issue
https://github.com/aws/aws-iot-device-sdk-embedded-C/issues/1880

### Testing
Built mqtt_mutual_auth demo without an issue. Executed demo successfully.

Commands to build
```
cmake -S . -B build -DAWS_IOT_ENDPOINT="myendpoint-ats.iot.us-east-1.amazonaws.com" -DCLIENT_CERT_PATH="/home/kstribrn/cool-certificate.pem.crt" -DCLIENT_PRIVATE_KEY_PATH="/home/kstribrn/my-privatekey.pem.key"
cd build
make mqtt_demo_mutual_auth
cd bin
./mqtt_demo_mutual_auth
```

Output
```
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:680] Establishing a TLS session to myendpoint-ats.iot.us-east-1.amazonaws.com:8883.
[INFO] [MQTT] [core_mqtt.c:2740] MQTT connection established with the broker.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1173] MQTT connection successfully established with broker.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1683] A clean MQTT connection is established. Cleaning up all the stored outgoing publishes.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1418] Subscribing to the MQTT topic testclient/example/topic.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1237] SUBSCRIBE sent for topic testclient/example/topic to broker.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1051] Subscribed to the topic testclient/example/topic. with maximum QoS 1.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1457] Sending Publish to the MQTT topic testclient/example/topic.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1335] PUBLISH sent for topic testclient/example/topic to broker with packet ID 2.


[INFO] [MQTT] [core_mqtt.c:1540] Ack packet deserialized with result: MQTTSuccess.
[INFO] [MQTT] [core_mqtt.c:1557] State record updated. New state=MQTTPublishDone.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1082] PUBACK received for packet id 2.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:792] Cleaned up outgoing publish packet with packet id 2.
...
```
By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -410,7 +410,8 @@ In order to set these configurations manually, edit `demo_config.h` in `demos/mq
 * Set `CLIENT_CERT_PATH` to the path of the client certificate downloaded when setting up the device certificate in [AWS IoT Account Setup](#aws-iot-account-setup).
 * Set `CLIENT_PRIVATE_KEY_PATH` to the path of the private key downloaded when setting up the device certificate in [AWS IoT Account Setup](#aws-iot-account-setup).
 
-It is possible to configure `ROOT_CA_CERT_PATH` to any PEM-encoded Root CA Certificate. However, this is optional because CMake will download and set it to [AmazonRootCA1.pem](https://www.amazontrust.com/repository/AmazonRootCA1.pem) when unspecified.
+It is possible to configure `ROOT_CA_CERT_PATH` to any PEM-encoded Root CA Certificate. However, this is optional because CMake will download and set it to [AmazonRootCA1.pem](https://www.amazontrust.com/repository/AmazonRootCA1.pem) when unspecified. If unspecified, the default Root CA
```
Comment (Skptak) [inline README.md:413]: Should split this line up so that it's not a single line that's 300 characters wide

File: README.md [hunk 2]
```diff
@@ -410,7 +410,8 @@ In order to set these configurations manually, edit `demo_config.h` in `demos/mq
...
+path will be interpreted relative to where the demo binary is executed. For many demos, you can change this path by modifying it in the corresponding `demo_config.h`.
```
Comment (AniruddhaKanhere) [inline README.md:414]: Or we should merge line 414 with 413 - just to be consistent.

File: README.md (hunk 1 above)
Comment (AniruddhaKanhere) [reply README.md:413]: I think doing that will make this single paragraph inconsistent from the whole readme. The whole readme is written with paragraphs on a single line.

Comment (kstribrnAmzn) [reply README.md]: I agree with your point Aniruddha. We may want to rework this README later. I'd prefer not to add to the problem.

Also - splitting these on separate lines doesn't lead to any changes in how the markdown is rendered, so I am fine splitting.

File: README.md (hunk 2 above)
Comment (kstribrnAmzn) [reply README.md]: I like this suggestion but will be going with the split so I don't add to the long line problem

**Review by AniruddhaKanhere — APPROVED** [review]

**Review by Skptak — APPROVED** [review]

**Pull request merged**